### DISCUSSION TITLE
Unable to generate desired request rate

### DISCUSSION THREAD

**Discussion by ankitkhedekar** [artilleryio/artillery#2271] [Artillery]
Hello,

I am trying to generate a load of around 300 request/sec to an API endpoint. From the documentation I understood this could be achieved by setting the `arrivalRate` to the desired value. But when I try that the `http.request_rate` at the max goes to about 190/sec
I tried running the same test on multiple configurations of gitlab runners (vCPUs 2,4,8,16) but unable to generate anywhere close to 300 requests/sec

Please suggest what could be possible issues here?

Using the `latest` artillery docker image here

**Comment by bernardobridge** [comment]
Hi @ankitkhedekar 👋 ,

Can you please share the test script and config that you're using to get to those values?

Thank you.

**Comment by hassy** [comment]
`arrivalRate` sets the number of virtual user arrivals, not the request rate. The actual request rate depends on several factors e.g.:

- How many actions each VU performs
- Response times for different requests from the server

Artillery does not support setting a constant request rate. We cover the reasoning behind that in this blog post: https://www.artillery.io/blog/load-testing-workload-models#:~:text=An%20open%20workload,is%20extremely%20narrow.

If you need to generate a constant request rate to a single endpoint, [autocannon](https://www.npmjs.com/package/autocannon) is a good option.
